cloud-media-livestream/keypublisher/clients/cpix_client.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""File containing abstract class for CPIX clients."""
+
+import abc
+import os
+
+from typing import Dict
+from typing import List
+
+from google.cloud import secretmanager
+
+
+class CpixClient(abc.ABC):
+    """Abstract class for CPIX clients."""
+
+    @abc.abstractmethod
+    def fetch_keys(self, media_id: str, key_ids: List[str]) -> Dict[str, object]:
+        """Fetches encryption keys and prepares JSON content to be written to Secret Manager.
+
+        Args:
+        media_id (string): Name for your asset, sometimes used by DRM providers to
+            show usage and reports.
+        key_ids (list[string]): List of IDs of any keys to fetch and prepare.
+
+        Returns:
+        Dictionary mapping key IDs to JSON-structured object containing key
+            information to be written to Secret Manager.
```

Comment:
Section items should be indented one level to the right, see https://google.github.io/styleguide/pyguide#383-functions-and-methods.
```suggestion
        Args:
            media_id (string): Name for your asset, sometimes used by DRM providers to
                show usage and reports.
            key_ids (list[string]): List of IDs of any keys to fetch and prepare.

        Returns:
            Dictionary mapping key IDs to JSON-structured object containing key
            information to be written to Secret Manager.
```

Reply:
Sounds good, indentation fixed for all files now.